web/src/features/panels/zone/ShareButton.tsx

```diff
@@ -20,68 +22,45 @@ interface ShareButtonProps
   showIosIcon?: boolean;
   hasMobileUserAgent?: boolean;
 }
+
 const trackShareClick = trackShare(ShareType.SHARE);
-const DURATION = 3 * 1000;
 
 export function ShareButton({
   iconSize = DEFAULT_ICON_SIZE,
-  shareUrl,
+  shareUrl = baseUrl,
   showIosIcon = isIos(),
   hasMobileUserAgent = isMobile(),
   ...restProps
 }: ShareButtonProps) {
   const { t } = useTranslation();
   const reference = useToastReference();
   const [toastMessage, setToastMessage] = useState('');
+  const { copyToClipboard, share } = useShare();
 
-  const url = shareUrl ?? window.location?.href;
-
-  const shareData = {
-    title: 'Electricity Maps',
-    text: 'Check this out!',
-    url,
-  };
-
-  const share = async () => {
-    try {
-      await CapShare.share(shareData);
-    } catch (error) {
-      if (error instanceof Error && !/AbortError|canceled/.test(error.toString())) {
-        console.error(error);
-        setToastMessage(t('share-button.share-error'));
-        reference.current?.publish();
-      }
-    }
-  };
-
-  const copyToClipboard = async () => {
-    try {
-      await navigator.clipboard.writeText(url);
-      setToastMessage(t('share-button.clipboard'));
-    } catch (error) {
-      console.error(error);
-      setToastMessage(t('share-button.clipboard-error'));
-    } finally {
+  const onClick = useCallback(async () => {
+    const toastMessageCallback = (message: string) => {
+      setToastMessage(message);
       reference.current?.publish();
-    }
-  };
+    };
 
-  const onClick = async () => {
     if (hasMobileUserAgent && (await CapShare.canShare())) {
-      share();
+      share(
+        {
+          title: 'Electricity Maps',
+          text: 'Check this out!',
```

Comment:
Maybe more of a question than feedback, but should we translate these?

Reply:
Yes, but we talked about potentially adding the dropdown to the share button, so I might lump fixing this copy & translations into that work